### PR TITLE
feat: support flat config

### LINKED ISSUE #122: Support new ESLint config
[Configuration Files (New)](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new)

### PR DESCRIPTION
**Pull request by sxzz**
docs: https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new

closes #122

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -2,14 +2,29 @@
...
+ * @returns flat ESLint config.
+ *
+ * @see [Configuration Files (New)](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new)
+ */
+function defineFlatConfig(config) {
+  return config;
+}
+
 exports.defineConfig = defineConfig;
+exports.defineFlatConfig = defineFlatConfig;
```
Comment (Shinigami92) [inline src/index.js:30]: Isn't it possible to use just `defineConfig` and manage the rest fully via TypeScript?
I think we can use function overloads, so when pass a `ESLintFlatConfig[]` the return value will be `ESLintFlatConfig[]` and when you pass `ESLintConfig` the return value will be `ESLintConfig`

File: src/flat/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * @see [Configuration Files (New)](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new)
+ */
+export interface ESLintFlatConfig {
+  /**
+   * An array of glob patterns indicating the files that the configuration object should apply to. If not specified, the configuration object applies to all files.
+   *
+   * @see [Ignore Patterns](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new#excluding-files-with-ignores)
+   */
+  files?: string[];
+  /**
```
Comment (Shinigami92) [inline src/flat/index.d.ts:18]:
```suggestion
  files?: string[];

  /**
```

**Comment by Shinigami92** [comment]
@sxzz I'm not sure about the term `legacy` 🤔 
Is it coming from ESLint? Or did you came up with this term?

**Comment by sxzz** [comment]
@Shinigami92 According to [the blog](https://eslint.org/blog/2022/08/new-config-system-part-3/#conclusion), I think flat config is the future. Actually, I'm not sure, but it has to get a name. What name do you prefer?
`legacy`, `nested`, or something else?

**Comment by Shinigami92** [comment]
> @Shinigami92 According to [the blog](https://eslint.org/blog/2022/08/new-config-system-part-3/#conclusion), I think flat config is the future. Actually, I'm not sure, but it has to get a name. What name do you prefer? `legacy`, `nested`, or something else?

Ah yeah, then I'm against `legacy` as it indicates **now** as it would be deprecated, but it is not. The new FlatConfig is just the new way for now.

For now I wouldn't change it at all and just stick with `config` and `flat-config` and use this terminology the whole way down.
Like `ESLintConfig` and `FlatESLintConfig` and so on.

... And oh yeah, please use `FlatESLintConfig` instead of `ESLintFlatConfig` as this is what they call it:

image

**Comment by sxzz** [comment]
@Shinigami92 I agree with you. I will change it

**Comment by Shinigami92** [comment]
I have some Meetings and Company Work today, if I find the time this evening, I will release your (welcome 🤗) additions 👍 
(I'm in German TimeZone)

**Comment by sxzz** [comment]
@Shinigami92 OK thanks~

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**